client-coroutines/src/test/kotlin/io/provenance/client/coroutines/PbClientTest.kt

```diff
@@ -27,4 +39,35 @@ class PbClientTest {
             )
         }
     }
+
+    @Test
+    fun `Simulated BROADCAST_MODE_BLOCK works`() = runTest {
```

Comment:
This whole test is ignored, but I tested this locally against testnet and it worked. It is out of scope for me to go through and put an actual integration test framework in place for this addition

Reply:
Were you also able to verify SYNC mode continues to work with the new implementation? Seems likely with the fall-through logic, but best to be certain ;)

Reply:
I tested that briefly by making sure it failed my assertions when switching, but good idea, I just tossed tests for SYNC mode in as well to show that